scripts/hooks/pre-commit

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+#-------------------------------------------------------------------------------
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#-------------------------------------------------------------------------------
+#
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+
+RETURN="0"
+SOURCE_FILES_PATTERNS=(
+  # Exclude documentation.
+  ':!*.md'
+  ':!**/README'
+
+  # Exclude data/input files.
+  ':!*.arcs'
+  ':!*.facts'
+  ':!*.csv'
+  ':!*.txt'
+
+  # Exclude test files.
+  ':!*.authlogic'
+  ':!*.raksha'
+  ':!*.ir'
+
+  # Exclude (generated) build and config files.
+  ':(exclude,glob).*'
+  ':!*.bazel'
+  ':!*.yaml'
+  ':(exclude,top)WORKSPACE'
+  ':(exclude,top)Dockerfile'
+
+  # Exclude special case files and directories.
+  ':(exclude,top)build_defs'
+  ':(exclude,top)third_party'
+  ':(exclude,top)images'
+  ':(exclude,top)gcb'
+  ':(exclude,top)rust'
+)
+
+error() {
+  RETURN="1"
+  echo -e "$1"
+}
+
+check_file_for_missing_license() {
+  if [ -d $file ]; then
+    return # This is a directory
+  fi
+  header=$(grep -E "https?:.*LICENSE" $file)
```

Comment:
This assumes that any license header will include a link to the actual license... this may not be a good assumption, but seemed to work for our code so far.